js/adapt-offlineStorage-xapi.js

```diff
@@ -0,0 +1,87 @@
+define([
+  'core/js/adapt',
+  './adapt-contrib-xapi'
+], function(Adapt, xapi) {
+
+  //xAPI handler for Adapt.offlineStorage interface.
+
+  // Use a lightweight fake model to pass into xAPI.sendState
+  var fakeModel = {
+    get: function() {
+      return 'offlineStorage';
+    }
+  };
+
+  var store = new Backbone.Model();
+  var isDataRestored = false;
+
+  Adapt.offlineStorage.initialize({
+
+    get: function(name) {
+      if (!name) {
+        return this.getAll();
+      }
+
+      if (!this.useTemporaryStore() && name.toLowerCase() === 'learnerinfo') {
+        return this.getLearnerInfo();
+      }
+
+      return store.get(name);
+    },
+
+    getAll: function() {
+      if (!isDataRestored) {
+        var state = xapi.get('state') || {};
+        store.set(state.offlineStorage);
+        isDataRestored = true;
+      }
+
+      //If not connected return just the store.
+      if (this.useTemporaryStore()) {
+        return store.toJSON();
+      }
+
+      return _.extend(store.toJSON(), {
+        learnerInfo: this.getLearnerInfo()
+      });
+    },
+
+    set: function(name, value) {
+      store.set(name, value);
+
+      // xAPI may not yet be initialised so use a soft trigger rather than hard calling xAPI.sendState
+      Adapt.trigger('state:change', fakeModel, store.toJSON());
+    },
+
+    useTemporaryStore: function() {
+      var config = Adapt.config.get('_xapi') || {};
+
+      return (!window.xapiWrapper || !config._isEnabled);
+    },
+
+    /**
+     * @returns {{name: string in the format Firstname Lastname, firstname: string, lastname: string }}
+     */
+    getLearnerInfo: function() {
+      var name = xapi.getLRSAttribute('actor').name || '';
```

Comment:
@danielstorey, I don't think you can rely on the `getLRSAttribute()` call here if the launch method was ADL.

It's better to use the `actor` property on the xAPI model.